lib/src/executables/react16_upgrade.dart

```diff
@@ -28,29 +30,84 @@ const _changesRequiredOutput = """
+  } else if (pubspecYaml.containsKey('dev_dependencies') &&
+      pubspecYaml['dev_dependencies'].containsKey(packageName)) {
+    return VersionConstraint.parse(
+        pubspecYaml['dev_dependencies'][packageName]);
+  }
+  return null;
+}
+
 void main(List<String> args) {
-  final query = FileQuery.dir(
-    pathFilter: isDartFile,
-    recursive: true,
-  );
-  exitCode = runInteractiveCodemodSequence(
-    query,
-    [
-      ReactDomRenderMigrator(),
-      ReactStyleMapsUpdater(),
-    ],
-    args: args,
-    defaultYes: true,
-    changesRequiredOutput: _changesRequiredOutput,
-  );
-
-  final logger = Logger('over_react_codemod.fixmes');
-  for (var dartFile in query.generateFilePaths()) {
-    final dartSource = File(dartFile).readAsStringSync();
-    if (dartSource.contains('[ ] $manualValidationCommentSubstring')) {
-      logger.severe(
-          'over_react_codemod validation comments are unaddressed in $dartFile');
-      exitCode = 1;
+  String constraintPackage;
+  VersionConstraint constraint;
+  final entrylogger = Logger('over_react_codemod.precheck');
+  try {
+    pubspecYaml = loadYaml(File('pubspec.yaml').readAsStringSync());
+  } catch (e) {
+    entrylogger.fine('This repo does not have a pubspec.yaml');
+    exitCode = 0;
+  }
+  try {
+    for (var package in packagesToCheckFor.entries) {
+      constraint = getDependencyVersion(package.key);
+      if (constraint != null) {
+        constraintPackage = package.key;
+        break;
+      }
+    }
+    if (constraint != null) {
+      if (!constraint.isAny &&
+          constraint.allowsAny(VersionConstraint.parse(
+              packagesToCheckFor[constraintPackage][0])) &&
+          constraint.allowsAny(VersionConstraint.parse(
+              packagesToCheckFor[constraintPackage][1]))) {
+        final query = FileQuery.dir(
+          pathFilter: isDartFile,
+          recursive: true,
+        );
+        exitCode = runInteractiveCodemodSequence(
+          query,
+          [
+            ReactDomRenderMigrator(),
+            ReactStyleMapsUpdater(),
+          ],
+          args: args,
+          defaultYes: true,
+          changesRequiredOutput: _changesRequiredOutput,
+        );
+
+        final logger = Logger('over_react_codemod.fixmes');
+        for (var dartFile in query.generateFilePaths()) {
+          final dartSource = File(dartFile).readAsStringSync();
+          if (dartSource.contains('[ ] $manualValidationCommentSubstring')) {
+            logger.severe(
+                'over_react_codemod validation comments are unaddressed in $dartFile');
+            exitCode = 1;
+          }
+        }
+      } else {
+        entrylogger
+            .fine('This repo does not appear to be in React 16 transition.');
+        exitCode = 0;
+      }
+    } else {
+      entrylogger.fine('This repo does not use React.');
+      exitCode = 0;
     }
+  } catch (e) {
+    entrylogger.fine(e);
+    exitCode = 0;
```

Comment:
There should be a comment here around why we're setting the exit code to 0 when in an error state